columnflow/tasks/histograms.py

```diff
@@ -117,12 +117,19 @@ def run(self):
         read_columns = {"category_ids", "process_id"} | set(aliases.values())
         read_columns |= {
             Route(variable_inst.expression)
-            if isinstance(variable_inst.expression, str) else
-            None  # TODO: handle variable_inst with custom expressions, can they declare columns?
             for variable_inst in (
                 self.config_inst.get_variable(var_name)
                 for var_name in law.util.flatten(self.variable_tuples.values())
             )
+            if isinstance(variable_inst.expression, str)
+        } | {
+            # for variable_inst with custom expressions, read columns declared via aux key
+            Route(inp)
+            for variable_inst in (
+                self.config_inst.get_variable(var_name)
+                for var_name in law.util.flatten(self.variable_tuples.values())
+            )
+            for inp in variable_inst.x("inputs", [])
```

Comment:
I'm not opposed to the double-comprehension, but it could be already worth to write a single loop over `variable_inst`'s and then fill `read_columns` in the body.

Reply:
True! Running over the `variable_inst`s twice is not that nice. In the end I refactored this as a single comprehension with two for loops, but if you prefer I can also convert it to a traditional for loop.